### PR TITLE
EES-5238 - breaking out shared ImportDataFunction and WriteDataFuncti…

### PR DESCRIPTION
**Pull request by duncan-at-hiveit**
This PR:
- extracts ImportData and WriteDataFiles into their own files for sharing with the new "Complete next data set version import" orchestration.
- extracts associated tests into their own files too.
- renames existing Process / CreateNextDataSetVersionFunction functions to mention "Mapping", as that is where the processing ends until the user completes the manual mappings.

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            .HandleFailuresOr(result => new OkObjectResult(result));
+    }
+
+    private async Task ProcessNextDataSetVersion(
+        DurableTaskClient client,
+        Guid dataSetVersionId,
+        Guid instanceId,
+        CancellationToken cancellationToken)
+    {
+        const string orchestratorName = nameof(ProcessNextDataSetVersionMappingsFunction.ProcessNextDataSetVersion);
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs:63]: Isn't this the orchestration that processes the next data set version to create the mappings?

I thought there'd need to be a second orchestration added to do the completion.

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ProcessNextDataSetVersionMappingsFunction.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
 
-public class ProcessNextDataSetVersionFunction(
+public class ProcessNextDataSetVersionMappingsFunction(
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ProcessNextDataSetVersionMappingsFunction.cs:12]: I think L16 and L17 below this need renaming from `ProcessNextDataSetVersion`  to `ProcessNextDataSetVersionMappings` to correspond with the renaming here and the renaming you've already done in the tests.

In the tests the inner class has been renamed to `ProcessNextDataSetVersionMappingsTests`.

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ActivityNames.cs [hunk 3]
```diff
@@ -3,12 +3,12 @@ namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Funct
...
-    public const string ImportMetadata = nameof(ImportMetadataFunction.ImportMetadata);
+    public const string ImportData = nameof(ImportDataFunction.ImportData);
+    public const string WriteDataFiles = nameof(WriteDataFilesFunction.WriteDataFiles);
     public const string HandleProcessingFailure = nameof(HandleProcessingFailureFunction.HandleProcessingFailure);
 
-    public const string ImportData = nameof(ProcessInitialDataSetVersionFunction.ImportData);
-    public const string WriteDataFiles = nameof(ProcessInitialDataSetVersionFunction.WriteDataFiles);
-    public const string CompleteInitialDataSetVersionProcessing = 
+    public const string ImportMetadata = nameof(ImportMetadataFunction.ImportMetadata);
+    public const string CompleteInitialDataSetVersionMappingProcessing =
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ActivityNames.cs:11]: NOOOOOOOOO.  I've changed it back, sorry!

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ActivityNames.cs [hunk 4]
```diff
@@ -3,12 +3,12 @@ namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Funct
...
     public const string CopyCsvFiles = nameof(CopyCsvFilesFunction.CopyCsvFiles);
-    public const string ImportMetadata = nameof(ImportMetadataFunction.ImportMetadata);
+    public const string ImportData = nameof(ImportDataFunction.ImportData);
+    public const string WriteDataFiles = nameof(WriteDataFilesFunction.WriteDataFiles);
     public const string HandleProcessingFailure = nameof(HandleProcessingFailureFunction.HandleProcessingFailure);
 
-    public const string ImportData = nameof(ProcessInitialDataSetVersionFunction.ImportData);
-    public const string WriteDataFiles = nameof(ProcessInitialDataSetVersionFunction.WriteDataFiles);
-    public const string CompleteInitialDataSetVersionProcessing = 
+    public const string ImportMetadata = nameof(ImportMetadataFunction.ImportMetadata);
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ActivityNames.cs:10]: Yup agreed, my bad.

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
+
+public class CompleteNextDataSetVersionProcessingFunction(
+    ILogger<CompleteNextDataSetVersionProcessingFunction> logger,
+    IDataSetVersionService dataSetVersionService,
+    IValidator<NextDataSetVersionCreateRequest> requestValidator)
+{
+    [Function(nameof(CreateNextDataSetVersion))]
+    public async Task<IActionResult> CreateNextDataSetVersion(
+        [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = nameof(CreateNextDataSetVersion))] [FromBody]
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs:24]: This one shouldn't even be in this PR sorry.

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs (hunk 1 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs:63]: As before, not supposed to be in this PR sorry

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ProcessNextDataSetVersionMappingsFunction.cs (hunk 2 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ProcessNextDataSetVersionMappingsFunction.cs:12]: Cool yup thanks, renamed this method now

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/ProcessorFunctionsIntegrationTest.cs [hunk 6]
```diff
@@ -192,11 +192,13 @@ protected override IEnumerable<Type> GetFunctionTypes()
         [
             typeof(CreateDataSetFunction),
             typeof(ProcessInitialDataSetVersionFunction),
-            typeof(CreateNextDataSetVersionFunction),
+            typeof(CreateNextDataSetVersionMappingsFunction),
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/ProcessorFunctionsIntegrationTest.cs]: I've removed this from the PR now

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs:24]: Removed from PR.

File: src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs (hunk 1 above)
Comment (duncan-at-hiveit) [reply src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/CompleteNextDataSetVersionProcessingFunction.cs:63]: Removed from PR.

**Review by benoutram — APPROVED** [review]

**Pull request merged**